### DISCUSSION TITLE
trigger on signUp - how to insert row in users_cart table after user confim email?

### DISCUSSION THREAD

**Discussion by nicitaacom** [supabase/supabase#18361] [Questions]
I know from docs that I can do it somehow like this
```sql
-- inserts a row into public.profiles
create function public.handle_new_user()
returns trigger
language plpgsql
security definer set search_path = public
as $$
begin
  insert into public.profiles (id)
  values (new.id);
  return new;
end;
$$;

-- trigger the function every time a user is created
create trigger on_auth_user_created
  after insert on auth.users
  for each row execute procedure public.handle_new_user();

```

But when I tried to create a triegger it no way to modify it

![image](https://github.com/supabase/supabase/assets/39565703/9ab6e65b-2fd6-4a80-83e6-6f13e93aeb5b)

So I need help with inserting a row in users_cart table (I do it only on user comfim email - not signUp)

**Comment by GaryAustin1** [comment]
https://supabase.com/docs/guides/database/postgres/triggers
Has more info on triggers. 
The UI will not let you edit the trigger (but will the function code in the function UI).  The trigger is pretty simple so should not normally need to  much editing.   You can also use the SQL editor with CREATE OR REPLACE and do the operation in SQL.  https://www.postgresql.org/docs/current/sql-createtrigger.html

**Reply by nicitaacom** [reply]
Yes but there are only 1 example
![image](https://github.com/supabase/supabase/assets/39565703/8c40bb52-c084-4d9b-822e-06570987cc90)


I want supabase to create example for default auth behavior - so I can insert row in users table when user verified email
If you know how to write this trigger - send me code that I can page in SQL edior in supabase

**Reply by GaryAustin1** [reply]
You showed that type of function in your first post...
https://supabase.com/docs/guides/auth/managing-user-data#using-triggers
If you don't want a profile table you can change that function to do whatever you want.
These are all just Postgres SQL functions so there is lots of info on the web with examples of code.

**Reply by nicitaacom** [reply]
Send me at least one of 'lots' of info in the WEB

![image](https://github.com/supabase/supabase/assets/39565703/0bb2f7ed-ac2f-4bfc-a8d2-1fce8676da68)


How to insert this row not only in auth table?

I don't see any trigger that happens when I login that insert row in auth schema 'users' table

**Reply by GaryAustin1** [reply]
Your very first example code at top is a trigger function on auth.users updating a profile table.  You can add more inserts on other tables in that same function.  That function runs when a user is first created and sets up a row in the profile table for them.   They have not confirmed their email at that point if that is important to you.  
 
Here was an example of someone using an UPDATE trigger function on auth.users to update another table.  They are specifically checking for the email confirmed getting set in auth.users in the trigger.  This code has to be run in the SQL editor.

https://github.com/orgs/supabase/discussions/15692
Note Their code had an error in it for not having 'security definer' so that would need to be fixed like your code you show up top.

As far as the web, I mean there are lots of examples of using SQL to insert into tables and writing trigger functions.  Specific examples of a trigger function on auth.users seem to be out there too but you already have an insert trigger on auth.users in your first post and I just linked an example of an update trigger on auth.users.
